packages/block-library/src/navigation/view.js

```diff
@@ -49,25 +49,18 @@ const { state, actions } = store(
 		},
 		actions: {
 			openMenuOnHover() {
-				const { type, overlayOpenedBy } = getContext();
-				if (
-					type === 'submenu' &&
-					// Only open on hover if the overlay is closed.
-					Object.values( overlayOpenedBy || {} ).filter( Boolean )
-						.length === 0
-				) {
+				const { overlayOpenedBy } = getContext();
+				// Opens only if the overlay is closed.
+				if ( Object.values( overlayOpenedBy || {} ).some( Boolean ) ) {
 					actions.openMenu( 'hover' );
 				}
 			},
 			closeMenuOnHover() {
-				const { type, overlayOpenedBy, submenuOpenedBy } = getContext();
+				const { overlayOpenedBy, submenuOpenedBy } = getContext();
+				// Closes only if not opened by click and the overlay is closed.
 				if (
-					type === 'submenu' &&
-					// Close only if not opened by click.
 					submenuOpenedBy.click === false &&
-					// Only close on hover if the overlay is closed.
-					Object.values( overlayOpenedBy || {} ).filter( Boolean )
-						.length === 0
+					Object.values( overlayOpenedBy || {} ).some( Boolean )
```

Comment:
This isn’t necessary but should introduce no behavioral differences. The `type === 'submenu'` conditions were useless because these handlers are only invoked by submenus (not the overlay). The `.filter( Boolean ).length === 0` is basically a more verbose version of `.some( Boolean )` which replaces it. Other than that this just revises the comments a bit.